src/gui/MixerView.cpp

```diff
@@ -385,48 +387,121 @@ void MixerView::deleteChannel(int index)
 	// can't delete master
 	if( index == 0 ) return;
 
-	// remember selected line
-	int selLine = m_currentMixerLine->channelIndex();
-
-	// in case the deleted channel is soloed or the remaining
-	// channels will be left in a muted state
-	Engine::mixer()->clearChannel(index);
-
-	// delete the real channel
-	Engine::mixer()->deleteChannel(index);
-
-	// delete the view
-	chLayout->removeWidget(m_mixerChannelViews[index]->m_mixerLine);
-	m_racksLayout->removeWidget( m_mixerChannelViews[index]->m_rackView );
-	delete m_mixerChannelViews[index]->m_fader;
-	delete m_mixerChannelViews[index]->m_muteBtn;
-	delete m_mixerChannelViews[index]->m_soloBtn;
-	// delete mixerLine later to prevent a crash when deleting from context menu
-	m_mixerChannelViews[index]->m_mixerLine->hide();
-	m_mixerChannelViews[index]->m_mixerLine->deleteLater();
-	delete m_mixerChannelViews[index]->m_rackView;
-	delete m_mixerChannelViews[index];
-	m_channelAreaWidget->adjustSize();
-
-	// make sure every channel knows what index it is
-	for(int i=index + 1; i<m_mixerChannelViews.size(); ++i)
+	// if channel is "used", confirm that the user want to delete the channel anyway
+	if (confirmRemoval(index))
```

Comment:
I think you can invert the condition and return early.

Reply:
ok.